deliver/spec/upload_screenshots_spec.rb

```diff
@@ -250,9 +252,29 @@
                               get_app_screenshot_sets: [app_screenshot_set])
         iterator = Deliver::AppScreenshotIterator.new([localization])
 
+        allow(Time).to receive(:now).and_return(0)
+
         expect_any_instance_of(Object).to receive(:sleep).with(kind_of(Numeric)).once
         expect(app_screenshot).to receive(:asset_delivery_state).and_return({ 'state' => 'UPLOAD_COMPLETE' }, { 'state' => 'COMPLETE' })
-        expect(subject.wait_for_complete(iterator)).to eq('COMPLETE' => 1)
+        expect(subject.wait_for_complete(iterator, 3600)).to eq('COMPLETE' => 1)
+      end
+    end
+
+    context 'when timeout is exceeded' do
+      it 'should exit the loop and return the current states' do
+        app_screenshot = double('Spaceship::ConnectAPI::AppScreenshot', asset_delivery_state: { 'state' => 'UPLOAD_COMPLETE' })
+        app_screenshot_set = double('Spaceship::ConnectAPI::AppScreenshotSet',
+                                    screenshot_display_type: Spaceship::ConnectAPI::AppScreenshotSet::DisplayType::APP_IPHONE_55,
+                                    app_screenshots: [app_screenshot])
+        localization = double('Spaceship::ConnectAPI::AppStoreVersionLocalization',
+                              locale: 'en-US',
+                              get_app_screenshot_sets: [app_screenshot_set])
```

Comment:
Just a heads up: generally I'd say that setting up mocks/doubles/etc should be part of a e.g. `before` block, and not live inside the `it` block 😅 but I see that you followed the style convention used throughout this file and for sake of consistency we can leave this as is :) thank you!

Reply:
Indeed, I'm not that experienced Ruby developer, so I've decided to follow the coding style around as closely as possible :)

Reply:
Great rationale! 😊